alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestIndexMonitorAction.kt

```diff
@@ -79,6 +82,33 @@ class RestIndexMonitorAction : BaseRestHandler() {
         val xcp = request.contentParser()
         ensureExpectedToken(Token.START_OBJECT, xcp.nextToken(), xcp)
         val monitor = Monitor.parse(xcp, id).copy(lastUpdateTime = Instant.now())
+
+        val monitorType = monitor.monitorType
+        val triggers = monitor.triggers
+        when (monitorType) {
+            Monitor.MonitorType.QUERY_LEVEL_MONITOR -> {
+                triggers.forEach {
+                    if (it !is QueryLevelTrigger) {
+                        throw IllegalArgumentException("Illegal trigger type")
+                    }
+                }
+            }
+            Monitor.MonitorType.BUCKET_LEVEL_MONITOR -> {
+                triggers.forEach {
+                    if (it !is BucketLevelTrigger) {
+                        throw IllegalArgumentException("Illegal trigger type")
+                    }
+                }
+            }
+            Monitor.MonitorType.DOC_LEVEL_MONITOR -> {
+                triggers.forEach {
+                    if (it !is DocumentLevelTrigger) {
+                        throw IllegalArgumentException("Illegal trigger type")
```

Comment:
For the error messages in the exceptions, make them more descriptive. For example, the Doc level one can be like: `throw IllegalArgumentException("Illegal trigger type, ${it.javaClass.name}, for Doc level monitor")`